// @ngx-lite/nav-drawer/esm2015/ngx-nav-drawer.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { NgxNavDrawerComponent } from './ngx-nav-drawer.component';
import * as i0 from "@angular/core";
export class NgxNavDrawerModule {
}
NgxNavDrawerModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.2", ngImport: i0, type: NgxNavDrawerModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
NgxNavDrawerModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.2", ngImport: i0, type: NgxNavDrawerModule, declarations: [NgxNavDrawerComponent], imports: [CommonModule], exports: [NgxNavDrawerComponent] });
NgxNavDrawerModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.2", ngImport: i0, type: NgxNavDrawerModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.2", ngImport: i0, type: NgxNavDrawerModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [NgxNavDrawerComponent],
                    exports: [NgxNavDrawerComponent],
                }]
        }] });
