// @ngx-lite/modal/esm2015/ngx-modal.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { NgxModalComponent } from './ngx-modal.component';
import * as i0 from "@angular/core";
export class NgxModalModule {
}
NgxModalModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.0.2", ngImport: i0, type: NgxModalModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
NgxModalModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.0.2", ngImport: i0, type: NgxModalModule, declarations: [NgxModalComponent], imports: [CommonModule], exports: [NgxModalComponent] });
NgxModalModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.0.2", ngImport: i0, type: NgxModalModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.0.2", ngImport: i0, type: NgxModalModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [NgxModalComponent],
                    exports: [NgxModalComponent],
                }]
        }] });
